### PR TITLE
Fix #560 AHC extension

### LINKED ISSUE #560: AHC extension
Plz. assign to me.

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

File: extensions/ahc/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+name: "Camel Quarkus HTTP Common"
+description: "Camel Quarkus HTTP Common"
```
Comment (lburgazzoli) [inline extensions/ahc/runtime/src/main/resources/META-INF/quarkus-extension.yaml]: looks like the name and description are wrong

File: extensions/pom.xml [hunk 2]
```diff
@@ -100,6 +100,7 @@
         <module>vm</module>
         <module>xslt</module>
         <module>zipfile</module>
+        <module>ahc</module>
```
Comment (lburgazzoli) [inline extensions/pom.xml]: nor ordered

File: pom.xml [hunk 3]
```diff
@@ -40,9 +40,11 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
+        <ahc.version>2.10.4</ahc.version>
```
Comment (lburgazzoli) [inline pom.xml]: do we need this version or can we inherit from camel-ahc ?

File: poms/bom-deployment/pom.xml [hunk 4]
```diff
@@ -35,7 +35,6 @@
     <properties>
         <camel-quarkus.version>1.1.0-SNAPSHOT</camel-quarkus.version><!-- kept in sync with project.version by the release plugin -->
 
-        <jandex-maven-plugin.version>1.0.6</jandex-maven-plugin.version>
```
Comment (lburgazzoli) [inline poms/bom-deployment/pom.xml]: this jandex thing seems unrelated to the pr, can we make this change in a separate commit ?

File: extensions/ahc/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 1 above)
Comment (ppalaga) [reply extensions/ahc/runtime/src/main/resources/META-INF/quarkus-extension.yaml]: +1

File: extensions/pom.xml (hunk 2 above)
Comment (ppalaga) [reply extensions/pom.xml]: +1

File: pom.xml (hunk 3 above)
Comment (ppalaga) [reply pom.xml]: Actually not for this PR directly, but it will be needed for the Telegram extension that needs AHC but has nothing to do with camel-ahc.

File: poms/bom-deployment/pom.xml (hunk 4 above)
Comment (ppalaga) [reply poms/bom-deployment/pom.xml]: Uneeded indeed. Let me remove it.

File: pom.xml (hunk 3 above)
Comment (lburgazzoli) [reply pom.xml]: yeah but this add some maintenance cost as we need to be sure this is in sync with the version used by camel

**Comment by ppalaga** [comment]
Luca's requests addressed in 58ca689

File: pom.xml (hunk 3 above)
Comment (ppalaga) [reply pom.xml]: Yes, that could be mitigated by 

(a) Importing the Camel BoM, or 
(b) Enabling the Dependency Convergence Rule on the Maven Enforcer plugin or
(c) A groovy script :)

**Comment by davsclaus** [comment]
ok to test

File: pom.xml (hunk 3 above)
Comment (lburgazzoli) [reply pom.xml]: I think we really need to consider option 1, eventually we may need to clean up a bit the camel's bom but I think that as long term we'll end up supporting most of the component camel supports and the chance to hit this issue again is in my opinion high

**Comment by lburgazzoli** [comment]
ok to test

**Comment by ppalaga** [comment]
88998b2 rebased and fixed conflicts
